framework/Databases.py

```diff
@@ -292,193 +293,186 @@ def __retrieveDataPointSet(self,attributes):
-            else:
-              inDict[key][i] = histVar[0][inputRow,ix]
+            inDict[key][i] = histVar[0][inputRow,ix]
           else:
             self.raiseAnError(IOError,'the parameter ' + key + ' has not been found in '+str(histVar[1]))
-      # outputPivotVal end case => PointSet is at the final status
-      if outputPivotValEnd:
-        if allOutParam:
-          for key in histVar[1]['outputSpaceHeaders']:
-            if i == 0:
-              outDict[key] = np.zeros(len(histList))
-            outDict[key][i] = histVar[0][-1,histVar[1]['outputSpaceHeaders'].index(key)]
-        else:
-          for key in outParam:
-            if key in histVar[1]['outputSpaceHeaders'] or \
-               utils.toBytes(key) in histVar[1]['outputSpaceHeaders']:
-              if i == 0:
-                outDict[key] = np.zeros(len(histList))
-              if key in histVar[1]['outputSpaceHeaders']:
-                outDict[key][i] = histVar[0][-1,histVar[1]['outputSpaceHeaders'].index(key)]
-              else:
-                outDict[key][i] = histVar[0][-1,histVar[1]['outputSpaceHeaders'].index(utils.toBytes(key))]
-            else:
-              self.raiseAnError(IOError,"the parameter " + str(key) + " has not been found")
-      elif outputRow != None:
-        if outputRow > histVar[0][:
-          ,0].size-1  and outputRow != -1: self.raiseAnError(IOError,'outputRow is greater than number of actual rows in Database '+ str(self.name) + '!')
-        if allOutParam:
-          for key in histVar[1]['outputSpaceHeaders']:
-            if i == 0:
-              outDict[key] = np.zeros(len(histList))
-            outDict[key][i] = histVar[0][outputRow,histVar[1]['outputSpaceHeaders'].index(key)]
-        else:
-          for key in outParam:
-            if key in histVar[1]['outputSpaceHeaders'] or \
-               utils.toBytes(key) in histVar[1]['outputSpaceHeaders']:
-              if i == 0:
-                outDict[key] = np.zeros(len(histList))
-              if key in histVar[1]['outputSpaceHeaders']:
-                outDict[key][i] = histVar[0][outputRow,histVar[1]['outputSpaceHeaders'].index(key)]
-              else:
-                outDict[key][i] = histVar[0][outputRow,histVar[1]['outputSpaceHeaders'].index(utils.toBytes(key))]
-            else:
-              self.raiseAnError(IOError,"the parameter " + str(key) + " has not been found")
-      elif operator != None:
+
+      if operator is not None:
         if operator not in ['max','min','average']:
-          self.raiseAnError(IOError,'operator unknown. Available are min,max,average')
+          self.raiseAnError(IOError,'operator "%s" unknown. Available are min,max,average'.format(operator))
         if histVar[1]['outputSpaceHeaders']:
           for key in histVar[1]['outputSpaceHeaders']:
             if i == 0:
               outDict[key] = np.zeros(len(histList))
+
+            keyIndex = histVar[1]['outputSpaceHeaders'].index(key)
```

Comment:
remove empty lines before and after this line

Reply:
done